packages/fx-core/src/common/local/projectSettingsHelper.ts

```diff
@@ -12,20 +17,24 @@ export class ProjectSettingsHelper {
       (pluginName) => pluginName === ResourcePlugins.SPFx
     );
 
-  public static includeFrontend = (projectSettings: ProjectSettings | undefined): boolean =>
-    !!(projectSettings?.solutionSettings as AzureSolutionSettings)?.activeResourcePlugins?.some(
-      (pluginName) => pluginName === ResourcePlugins.FrontendHosting
-    );
-
+  public static includeFrontend = (projectSettings: ProjectSettings | undefined): boolean => {
+    if (!projectSettings) return false;
+    const solutionSettings = projectSettings.solutionSettings as AzureSolutionSettings;
```

Comment:
Will `solutionSettings` or `capabilities` be undefined?

Reply:
No, their are required